library/src/main/java/com/schibsted/spain/barista/internal/matcher/BackgroundMatcher.kt

```diff
@@ -0,0 +1,65 @@
+package com.schibsted.spain.barista.internal.matcher
+
+import android.graphics.Bitmap
+import android.graphics.Canvas
+import android.graphics.drawable.Drawable
+import android.support.annotation.DrawableRes
+import android.view.View
+import android.widget.ImageView
+import com.schibsted.spain.barista.internal.util.BitmapComparator
+import com.schibsted.spain.barista.internal.util.DrawableToBitmapConverter
+import org.hamcrest.Description
+import org.hamcrest.TypeSafeMatcher
+
+class BackgroundMatcher private constructor(@DrawableRes private val expectedDrawableRes: Int) : TypeSafeMatcher<View>(View::class.java) {
+
+  companion object {
+    private const val EMPTY = -1
+    private const val ANY = -2
+
+    @JvmStatic
+    fun withBackground(@DrawableRes resourceId: Int) = BackgroundMatcher(resourceId)
+
+    @JvmStatic
+    fun withAnyBackground() = BackgroundMatcher(ANY)
+
+    @JvmStatic
+    fun withoutBackground() = BackgroundMatcher(EMPTY)
+  }
+
+  private var resourceName: String? = null
+
+  override fun matchesSafely(target: View): Boolean {
+    val view = target
+    if (expectedDrawableRes == EMPTY) {
+      return view.background == null
+    }
+    if (expectedDrawableRes == ANY) {
+      return view.background != null
+    }
+    if (view.background == null){
+      return false
+    }
+    val resources = target.context.resources
+    val expectedDrawable = resources.getDrawable(expectedDrawableRes)
+    resourceName = resources.getResourceEntryName(expectedDrawableRes)
+
+    if (expectedDrawable == null) {
+      return false
+    }
+
+    val bitmap = DrawableToBitmapConverter.getBitmap(view.background)
+    val otherBitmap = DrawableToBitmapConverter.getBitmap(expectedDrawable)
+    return BitmapComparator.compare(bitmap, otherBitmap)
+  }
+
+  override fun describeTo(description: Description) {
+    description.appendText("with background from resource id: ")
+    description.appendValue(expectedDrawableRes)
```

Comment:
I don't think printing the resource id as an int gives any value. I suggest to remove it and just leave the resource name.

Reply:
Really? You can look for that ID in your generated R class. Information is power. And we were doing it on the previous versions of Barista, at the DrawableMatcher.

Reply:
Espresso also does it, so... ok!